lib/api/BackbeatAPI.js

```diff
@@ -0,0 +1,131 @@
+
+        this.xproducer = new Kafka.Producer({
+            'metadata.broker.list': this._kafkaHost,
+            'dr_cb': true,  // delivery report
+        });
+
+        this._zkClient = zookeeper.createClient(this._zkEndpoint);
+        this._zkClient.connect();
+    }
+
+    /**
+     * Check if incoming request is valid
+     * @param {string} route - request route
+     * @return {boolean} true/false
+     */
+    isValidRoute(route) {
+        // refactor this
+        const validPaths = ['/_/healthcheck', '/_/throughput'];
+        return validPaths.includes(route);
+    }
+
+    /**
+     * Check if Kafka, Zookeeper, Producer, Consumer are connected
+     * @return {boolean} true/false
+     */
+    isConnected() {
+        // TODO:
+        // Check if Kafka and Zookeeper are connected
+        // Check if any node has fewer ISRs than normal (?)
+        return this._consumer.isConnected() && this._producer.isConnected()
+            && (this._zkClient.getState().name === 'SYNC_CONNECTED');
+    }
+
+    /**
+     * Get Kafka healthcheck
+     * @param {BackbeatRequest} backbeatRequest - Backbeat request object
+     * @param {function} cb - callback for response
+     * @return {undefined}
+     */
+    healthcheck(backbeatRequest, cb) {
+        this._producer.getMetadata({}, (err, meta) => {
+            if (err) {
+                // setup backbeatRequest error properly
+                return cb(backbeatRequest, err);
+            }
+            const validMD = meta.topics.filter(m => m.name === this._topic);
+
+            // TODO: Does ISR have to match ReplicationFactor?
+            //       Below implementation is probably invalid.
+            //       Also: kafka `transaction.state.log.min.isr`
+
+            const validISR = validMD.every(node =>
+                node.partitions.every(part => part.isrs.length === 3));
+
+            if (!validISR) {
+                backbeatRequest.setError({
+                    error: 'Internal Error: minimum in-sync replicas not met',
+                    code: 500,
+                });
+                return cb(backbeatRequest, backbeatRequest.getError());
+            }
+
+            return cb(backbeatRequest, validMD);
+        });
+    }
+
+    /**
+     * Get throughput for topic consumption
+     * @param {BackbeatRequest} backbeatRequest - Backbeat request object
+     * @param {function} cb - callback for response
+     * @return {undefined}
+     */
+    throughput(backbeatRequest, cb) {
+        const testTopic = 'hai-backbeat';
+        const value = new Buffer('hai I am producing this msg');
+        const key = 'key-1';
+        const partition = -1;
+
+        this._producer.produce(testTopic, partition, value, key);
+        this._consumer.subscribe([testTopic]);
+
+        const start = Date.now();
+        let end;
+        this._consumer.consume(1, () => {
+            end = Date.now();
+
+            cb(backbeatRequest, { difference: end - start });
```

Comment:
I guess this is work in progress, still a few considerations:

 - `elapsedMs` would be a better name here
 - the function measures the latency of a single operation rather than throughput, which is not be that useful of a metric since it does not reflect real workload
 - the function still does a useful healthcheck, so the function may be renamed to `consumerHealthcheck` for example
 - for real testing we would want to know how many operations per second we can achieve on kafka, but this requires a proper test environment and a specific test function that is not a good fit for a generic API.

Reply:
Note that as a healthcheck, having the latency of one operation is useful to detect abnormally high latency, in case the system has a problem (but as said, not useful to really know what is the system performance).